internal/provider/models/alerts/base_model.go

```diff
@@ -0,0 +1,207 @@
+package alerts
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	. "github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	. "github.com/mezmo/terraform-provider-mezmo/internal/provider/models/modelutils"
+)
+
+type SchemaAttributes map[string]schema.Attribute
+
+var baseAlertSchemaAttributes = SchemaAttributes{
+	// Non-config fields
+	"id": schema.StringAttribute{
+		Computed:    true,
+		Description: "The uuid of the alert",
+	},
+	"pipeline_id": schema.StringAttribute{
+		Required:    true,
+		Validators:  []validator.String{stringvalidator.LengthAtLeast(1)},
+		Description: "The uuid of the pipeline",
+	},
+	"component_kind": schema.StringAttribute{
+		Required:    true,
+		Validators:  []validator.String{stringvalidator.LengthAtLeast(1)},
+		Description: "The kind of component that the alert is attached to",
+	},
+	"component_id": schema.StringAttribute{
+		Required:    true,
+		Validators:  []validator.String{stringvalidator.LengthAtLeast(1)},
+		Description: "The uuid of the component that the alert is attached to",
+	},
+	"inputs": schema.ListAttribute{
+		ElementType: types.StringType,
+		Optional:    true,
+		Computed:    true, // The server could set a default, so it's computed
+		Description: "The ids of the input components",
+	},
+	"active": schema.BoolAttribute{
+		Optional:    true,
+		Computed:    true,
+		Description: "Indicates if the alert is turned on or off",
+		Default:     booldefault.StaticBool(true),
+	},
+	// Alert config fields
+	"name": schema.StringAttribute{
+		Required:    true,
+		Description: "The name of the alert.",
+	},
+	"description": schema.StringAttribute{
+		Optional:    true,
+		Description: "An optional description describing what the alert is for.",
+	},
+	"event_type": schema.StringAttribute{
+		Required:    true,
+		Description: "The type of event is either a Log event or a Metric event.",
+	},
+	"group_by": schema.ListAttribute{
+		ElementType: StringType{},
+		Optional:    true,
+		Description: "When aggregating, group events based on matching values from each of " +
+			"these field paths. Supports nesting via dot-notation. This value is " +
+			"optional for Log event types, and SHOULD be used for Metric event types.",
+		// TODO: require this, even if it's empty? Because leaving it blank for a metric will defailt to [.host, .namespace...]
```

Comment:
We might wanna make this change now before we merge.  Overall, we should minimize the use of `.default()` in the schema because default values for certain fields can actually cause unknown behavior here.  For example, if we default `group_by` to `[.host, .namespace, .tags]` for a metric event, the user wouldn't know.

If we tune the schema such that most values need to be provided, it would probably work better.  The UI will supply everything anyway, so a heavy use of auto-assigned defaults doesn't make much sense.  Based on those changes, some of the descriptions and error messages would change here.